src/main/java/org/wildfly/security/sasl/digest/AbstractDigestMechanism.java

```diff
@@ -298,17 +298,18 @@ protected DigestWrapper(boolean confidential) {
         System.arraycopy(messageMac, 0, toCipher, len + paddingLength, 10);
 
         byte[] cipheredPart = null;
+        byte[] result = null;
         try {
             cipheredPart = wrapCipher.update(toCipher);
+
```

Comment:
This change just seems to be allowing the NullPointerException to be thrown and then wrap it later.

Instead I think we should detect and handle the null - there are very specific cases where it could be null so I think a message logged at trace plus an appropriate error may be better.